refactor: use structured logger in video adapters

pedroccastro opened the pull request:
## What does this PR do?

Replaces console statements with structured logger in Office365 (Teams) and Webex video adapters. This improves observability and aligns with the logging patterns used across the codebase.

### Changes

- Replaced `console.log`/`console.error` with `logger.debug`/`logger.error`
- Added debug logs for meeting lifecycle events (create, delete)
- Standardized error logging with structured format (message + name)
- Fixed error messages to use null-safe access and clearer descriptions

### How to test

1. Connect Office365 (Teams) or Webex integration
2. Create a booking with video conferencing
3. Check server logs for structured debug output:
     - `Creating Teams/Webex meeting`
     - `Teams/Webex meeting created { meetingId: ... }`
4. Verify no raw console output in production logs

## Mandatory Tasks

- [x] I have self-reviewed the code
- [ ] **N/A** I have updated the developer docs in /docs if this PR makes changes that would require a documentation change
- [ ] **N/A** I confirm automated tests are in place that prove my fix is effective or that my feature works.

pedroccastro committed:
refactor: use structured logger in video adapters

- Remove debug console.log/console.error statements
- Add logger.debug for observability (meeting lifecycle events)
- Add logger.error with safe error pattern (message + name only)
- Fix error messages to avoid exposing response details

pedroccastro committed:
Merge branch 'main' into refactor/video-adapters-cleanup

volnei commented on packages/app-store/webex/lib/VideoApiAdapter.ts:
```diff
@@ -1,6 +1,7 @@
 
 import dayjs from "@calcom/dayjs";
+import logger from "@calcom/lib/logger";
 import prisma from "@calcom/prisma";
```
```suggestion
import { prisma } from "@calcom/prisma";
```

default export is deprecated, since we're here

volnei commented on packages/app-store/webex/lib/VideoApiAdapter.ts:
```diff
@@ -171,8 +172,10 @@ const WebexVideoApiAdapter = (credential: CredentialPayload): VideoApiAdapter =>
-        console.error(err);
-
+        logger.error("Error fetching Webex availability", {
+          message: err instanceof Error ? err.message : String(err),
```
should we take care with this error message? to avoid exposing secrets etc? Maybe err.message is enought

volnei commented on packages/app-store/webex/lib/VideoApiAdapter.ts:
```diff
@@ -195,22 +198,27 @@ const WebexVideoApiAdapter = (credential: CredentialPayload): VideoApiAdapter =>
         const result = webexEventResultSchema.parse(response);
         if (result.id && result.webLink) {
+          logger.debug("Webex meeting created", { meetingId: result.id });
           return {
```
not sure if we need logger and throw, since all throws will be logged anyways

volnei commented on packages/app-store/webex/lib/VideoApiAdapter.ts:
```diff
@@ -195,22 +198,27 @@ const WebexVideoApiAdapter = (credential: CredentialPayload): VideoApiAdapter =>
+          message: err instanceof Error ? err.message : String(err),
+          name: err instanceof Error ? err.name : "UnknownError",
+        });
         throw new Error("Unexpected error");
```
same here.. should we just rethrow the error?

volnei reviewed (changes requested):
nit comments

pedroccastro committed:
fix: address review feedback on Webex video adapter

- Use named import for prisma (default export deprecated)
- Remove error details from logs to avoid exposing secrets
- Remove redundant try/catch blocks (errors propagate naturally)

pedroccastro committed:
Merge branch 'main' into refactor/video-adapters-cleanup

volnei reviewed (approved):
Great :rocket:
